common/tools/dev-tool/src/commands/admin/list/packages.ts

```diff
@@ -61,17 +66,25 @@ export async function getProjects(service?: string): Promise<RushJsonProject[]>
     : rushJson.projects;
 }
 
-export default leafCommand(commandInfo, async ({ paths, service }) => {
+export default leafCommand(commandInfo, async ({ paths, service, task }) => {
   const cwd = process.cwd();
   const root = await resolveRoot();
 
   const projects = await getProjects(service);
 
-  for (const project of projects) {
-    if (paths) {
-      console.log(path.relative(cwd, path.resolve(root, project.projectFolder)) || ".");
-    } else {
-      console.log(project.packageName);
+  if (task) {
+    const imported = await import(task);
+    console.dir(imported.default);
+    for (const project of projects) {
+      await imported.default(project, paths, cwd, root);
+    }
+  } else {
```

Comment:
One way to remove this conditional would be to have a task (maybe even the new echo-packages.js task) act as a default or fallback when a task is not provided. That could simplify a bit of the code here. Just a thought though, there are pros and cons of course so feel free to ignore!

Reply:
Good point!  Now falling back to a function with the original echo code.